integration-tests/matching.ts

```diff
@@ -35,6 +36,194 @@ void test('Pupil Request Match', async () => {
+});
+
+const studentWithMR = test('Student Request Match', async () => {
+    const { client, student } = await studentOne;
+
+    const { me: s } = await client.request(`
+    query GetOpenMatchRequestCount {
+        me {
+            student {
+                openMatchRequestCount
+            }
+        }
+     }
+    `);
+
+    // Match request is 0 after registration
+    assert.strictEqual(s.student.openMatchRequestCount, 0);
+
+    await client.request(`
+        mutation {
+            studentCreateMatchRequest
+        }
+    `);
+
+    const { me: s1 } = await client.request(`
+        query GetOpenMatchRequestCount {
+            me {
+                student {
+                    openMatchRequestCount
+                }
+            }
+        }
+    `);
+
+    assert.strictEqual(s1.student.openMatchRequestCount, 1);
+    return { client, student };
+});
+
+const match1 = test('Manual Match creation', async () => {
+    const { pupil, client: pupilClient } = await pupilWithMR;
+    const { student, client: studentClient } = await studentWithMR;
+
+    await adminClient.request(`
+        mutation CreateManualMatch {
+            matchAdd(poolName: "lern-fair-now", studentId: ${student.student.id} pupilId: ${pupil.pupil.id})
+        }
+    `);
+
+    const { me: { pupil: p1 }} = await pupilClient.request(`
+        query PupilWithMatch {
+            me {
+                pupil {
+                    openMatchRequestCount
+                    matches {
+                        id
+                        dissolved
+                        student { firstname lastname }
+                    }
+                }
+            }
+        }
+    `);
+
+    assert.strictEqual(p1.openMatchRequestCount, 0);
+    assert.strictEqual(p1.matches.length, 1);
+    assert.strictEqual(p1.matches[0].dissolved, false);
+    assert.strictEqual(p1.matches[0].student.firstname, student.firstname);
+    assert.strictEqual(p1.matches[0].student.lastname, student.lastname);
+
+    const { me: { student: s1 }} = await studentClient.request(`
+        query StudentWithMatch {
+            me {
+                student {
+                    openMatchRequestCount
+                    matches {
+                        id
+                        dissolved
+                        pupil { firstname lastname }
+                    }
+                }
+            }
+        }
+    `);
+
+    assert.strictEqual(s1.openMatchRequestCount, 0);
+    assert.strictEqual(s1.matches.length, 1);
+    assert.strictEqual(s1.matches[0].dissolved, false);
+    assert.strictEqual(s1.matches[0].pupil.firstname, pupil.firstname);
+    assert.strictEqual(s1.matches[0].pupil.lastname, pupil.lastname);
+
+    return { id: s1.matches[0].id, pupil, student, pupilClient, studentClient };
+});
+
+void test('Create Chat for Match', async () => {
+    const { id, pupil, student, pupilClient, studentClient } = await match1;
+
+    // Creating a Chat ensures all users have accounts:
```

Comment:
That's how the mocking would look like then:

Reply:
Are we mocking based on the url as identifier or is it mocked sequentially (so the first expectFetch mocks the first fetch call and so on)?